modules/in_band_dtmf/in_band_dtmf.c

```diff
@@ -0,0 +1,329 @@
+/**
+ * @file in_band_dtmf.c  DTMF decoder
+ */
+#include <time.h>
+#include <ctype.h>
+#include <string.h>
+#include <re.h>
+#include <rem.h>
+#include <baresip.h>
+
+
+/**
+ * @defgroup in_band_dtmf in_band_dtmf
+ *
+ * Audio filter that de- and encodes DTMF tones.
+ *
+ * The tone length of an encoded tone is 0.1s.
+ * New tones can be added while encoding is active.
+ */
+
+
+struct in_band_dtmf_filt_dec {
+	struct aufilt_enc_st af;  /* inheritance */
+	struct dtmf_dec *dec;
+	struct ua *ua;
+	struct call *call;
+	struct le le_priv;
+};
+
+
+struct in_band_dtmf_filt_enc {
+	struct aufilt_enc_st af;  /* inheritance */
+	struct mbuf *mb;
+	unsigned srate;
+	struct le le_priv;
+};
+
+
+static struct list encs;
+static struct list decs;
+
+
+static void in_band_dtmf_dec_handler(char digit, void *arg)
+{
+	(void)arg;
+	char key_str[2];
+	struct in_band_dtmf_filt_dec *st;
+
+	key_str[0] = digit;
+	key_str[1] = '\0';
+
+	if (!list_count(&decs)) {
```

Comment:
It is more efficient to use `list_isempty` here.

Reply:
Fixed by 5d151b85